packages/patternfly-4/react-core/src/components/Pagination/Navigation.js

```diff
@@ -40,99 +41,130 @@ const defaultProps = {
   onPageInput: () => undefined
 };
 
-const Navigation = ({
-  page,
-  lastPage,
-  pagesTitle,
-  toLastPage,
-  toNextPage,
-  toFirstPage,
-  toPreviousPage,
-  currPage,
-  paginationTitle,
-  onSetPage,
-  onNextClick,
-  onPreviousClick,
-  onFirstClick,
-  onLastClick,
-  onPageInput,
-  className,
-  ...props
-}) => (
-  <nav className={css(styles.paginationNav, className)} aria-label={paginationTitle} {...props}>
-    <Button
-      variant={ButtonVariant.plain}
-      isDisabled={page === 1}
-      aria-label={toFirstPage}
-      data-action="first"
-      onClick={event => {
-        onFirstClick(event, 1);
-        onSetPage(event, 1);
-      }}
-    >
-      <AngleDoubleLeftIcon />
-    </Button>
-    <Button
-      variant={ButtonVariant.plain}
-      isDisabled={page === 1}
-      data-action="previous"
-      onClick={event => {
-        const newPage = page - 1 >= 1 ? page - 1 : 1;
-        onPreviousClick(event, newPage);
-        onSetPage(event, newPage);
-      }}
-      aria-label={toPreviousPage}
-    >
-      <AngleLeftIcon />
-    </Button>
-    <div className={css(styles.paginationNavPageSelect)}>
-      <input
-        className={css(styles.formControl)}
-        aria-label={currPage}
-        type="number"
-        min="1"
-        max={lastPage}
-        value={page}
-        onChange={event => {
-          let inputPage = Number.parseInt(event.target.value, 10);
-          inputPage = Number.isNaN(inputPage) ? page : inputPage;
-          inputPage = inputPage > lastPage ? lastPage : inputPage;
-          inputPage = inputPage < 1 ? 1 : inputPage;
-          onSetPage(event, Number.isNaN(inputPage) ? page : inputPage);
-          onPageInput(event, Number.isNaN(inputPage) ? page : inputPage);
-        }}
-      />
-      <span aria-hidden="true">
-        of {lastPage} {pagesTitle}
-      </span>
-    </div>
-    <Button
-      variant={ButtonVariant.plain}
-      isDisabled={page === lastPage}
-      aria-label={toNextPage}
-      data-action="next"
-      onClick={event => {
-        const newPage = page + 1 <= lastPage ? page + 1 : lastPage;
-        onNextClick(event, newPage);
-        onSetPage(event, newPage);
-      }}
-    >
-      <AngleRightIcon />
-    </Button>
-    <Button
-      variant={ButtonVariant.plain}
-      isDisabled={page === lastPage}
-      aria-label={toLastPage}
-      data-action="last"
-      onClick={event => {
-        onLastClick(event, lastPage);
-        onSetPage(event, lastPage);
-      }}
-    >
-      <AngleDoubleRightIcon />
-    </Button>
-  </nav>
-);
+class Navigation extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { userInputPage: this.props.page };
+  }
+
+  parseInteger = (input, lastPage) => {
```

Comment:
Can we hoist this up outside the class into `const parseInteger = (input, lastPage) => {...}`?

Reply:
Why? Can it be used elsewhere? If so, we should move it to utilities.

Reply:
It's not reusable; I just broke it out here since it's been used in this class a couple of times. I made it private.